quark/subproject.py

```diff
@@ -285,11 +287,17 @@ def checkout(self):
                 extra_opts += ["--depth", "1"]
             # Needed essentially for the shallow case, as for full clones the
             # git clone -n + git checkout would suffice
-            if not self.ref_is_commit and self.ref != 'origin/HEAD':
+            if shallow and self.ref_type != 'commit' and self.ref != 'origin/HEAD':
```

Comment:
`if shallow` is unneeded: we are in the branch `if self.ref == 'commit' and shallow`

Reply:
You missed the `if` condition above.